pkg/cloud-provider/cloudapi/aws/aws_security.go

```diff
@@ -788,35 +791,36 @@ func (c *awsCloud) UpdateSecurityGroupMembers(groupIdentifier *securitygroup.Clo
 	}
 	ec2Service := serviceCfg.(*ec2ServiceConfig)
 
-	groupCloudSgName := groupIdentifier.GetCloudName(membershipOnly)
+	cloudSgName := securityGroupIdentifier.GetCloudName(membershipOnly)
 
 	// get addressGroup cloudSgID
 	vpcIDs := []string{vpcID}
-	cloudSgNames := map[string]struct{}{groupCloudSgName: {}}
+	cloudSgNames := map[string]struct{}{cloudSgName: {}}
 	out, err := ec2Service.getCloudSecurityGroupsWithNameFromCloud(vpcIDs, cloudSgNames)
 	if err != nil {
 		return err
 	}
 	if len(out) != len(cloudSgNames) {
 		return fmt.Errorf("failed to find cloud sg (%v) corresponding to address group (%v)",
-			groupIdentifier.Name, groupCloudSgName)
+			securityGroupIdentifier.Name, cloudSgName)
 	}
-	groupCloudSgID := out[groupCloudSgName].GroupId
+	cloudSgID := out[cloudSgName].GroupId
 
-	err = ec2Service.updateSecurityGroupMembers(groupCloudSgID, groupCloudSgName, vpcID, cloudResourceIdentifiers, membershipOnly)
+	err = ec2Service.updateSecurityGroupMembers(cloudSgID, cloudSgName, vpcID, cloudResourceIdentifiers, membershipOnly)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (c *awsCloud) DeleteSecurityGroup(groupIdentifier *securitygroup.CloudResource, membershipOnly bool) error {
+// DeleteSecurityGroup invokes cloud api and deletes the cloud security group. Any attached resource will be moved to default sg.
```

Comment:
Is this happening? Like VM moving to default SG.. Please validate all cases

Reply:
Yes this is happening. When last nephe SG is removed AWS VMs are moved to the VPC default SG

Reply:
Can you validate the same for Azure.. Also, can you add integration test for it... I want this behave to be consistent..